python/lsst/pipe/base/configOverrides.py

```diff
@@ -123,3 +140,16 @@ def applyTo(self, config):
                     # this can throw
                     value = eval(value, {})
                     setattr(obj, field[-1], value)
+            elif otype == 'namesDict':
+                try:
+                    parsedNamesDict = ast.literal_eval(override)
+                    if not isinstance(parsedNamesDict, dict):
+                        raise ValueError()
```

Comment:
Some message to user would be useful here

Reply:
Python code isn't meant to raise `pexExceptions.RuntimeError`.

Reply:
Regarding ValueError - I think it is more complicated. `literal_eval()` can certainly throw exceptions other than ValueError (SyntaxError is probably even more common). Also I'm not a big fan of chained exceptions as they can be confusing for end users (especially as in this case ValueError does not have any useful message) but I think we do not have any explicit policy against chaining. Still I think adding a message to ValueError here would be better.